test/Microsoft.AspNetCore.Server.KestrelTests/EngineTests.cs

```diff
@@ -878,11 +878,7 @@ public void ConnectionCanReadAndWrite(ListenOptions listenOptions)
                     throw onStartingException;
                 }, null);
 
-                response.Headers["Content-Length"] = new[] { "11" };
-
-                var writeException = await Assert.ThrowsAsync<ObjectDisposedException>(async () =>
-                    await response.Body.WriteAsync(Encoding.ASCII.GetBytes("Hello World"), 0, 11));
-
+                var writeException = await Assert.ThrowsAsync<ObjectDisposedException>(async () => await response.Body.FlushAsync());
```

Comment:
You do. The point here is to not only test that you get a 500 when an `OnStarting` callback throws, but also that the connection is not closed in that situation. Because the test code captures the exception from `WriteAsync` and does not rethrow it, we end up verifying the amount of bytes written after the app runs (see link above), which also leads to a 500 response but this one also closes the connection (because too few bytes were written and that will confuse the client).

Switching this to `FlushAsync()` makes the test simpler and avoids that situation. An alternative would be to keep the `WriteAsync` call and rethrow the exception, but this will result in more errors being logged and will require changing the assertion for that.